src/mpi/comm/commutil.c

```diff
@@ -567,6 +568,13 @@ int MPIR_Comm_commit(MPIR_Comm * comm)
     MPIR_Assert(comm->node_comm == NULL);
     MPIR_Assert(comm->node_roots_comm == NULL);
 
+    /* Every user-level communicator gets a unique vci sequence.
+     * It can be further hashed into actual vci that is defined at device layer.
+     */
+    static int vci_seq = 0;
+    comm->vci = vci_seq;
+    vci_seq++;
```

Comment:
What is the purpose of `vci_seq` again? Is it to ensure that the communicators on different ranks map to the same VCI?

Also, since it is not being used in this PR, we can defer this commit to a later PR where it is actually being used.

Reply:
You are right, we'll need collective synchronization across the communicator.

Reply:
Updated.